src/services/event/event-router.ts

```diff
@@ -64,15 +64,26 @@ eventsRouter.use(cors({ origin: "*" }));
  * @apiUse strongVerifyErrors
  * @apiError (403: Forbidden) {String} Forbidden Not a valid staff token.
  * */
-eventsRouter.get("/staff/", strongJwtVerification, async (_: Request, res: Response, next: NextFunction) => {
+eventsRouter.get("/staff/", strongJwtVerification, async (req: Request, res: Response, next: NextFunction) => {
     const payload: JwtPayload = res.locals.payload as JwtPayload;
 
     if (!hasStaffPerms(payload)) {
         return next(new RouterError(StatusCode.ClientErrorForbidden, "Forbidden"));
     }
 
-    const staffEvents: StaffEvent[] = await Models.StaffEvent.find();
-    return res.status(StatusCode.SuccessOK).send({ events: staffEvents });
+    const showAllShiftsParam: string | undefined = req.query.showAllShifts as string | undefined;
```

Comment:
also im only testing with my token (admin) but make sure that if non-admin staff roles request showAllShifts=True, it redirects them to just shifts assigned to them - they shouldnt be able to see everything